### PR TITLE
Implement common service to start up REST API

### PR DESCRIPTION
**Pull request by DanNiESh**
Run cmd: flocx-market-api, then the app is start by the script cmd/start_service.py
- TG-86

### CODE REVIEW

File: flocx_market/api/app.py [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
 
+my_bp = Blueprint("my_bp", __name__)
 
-def get_pecan_config():
-    filename = api_config.__file__.replace('.pyc', '.py')   # get the absolute path of the pecan config.py
-    return pecan.configuration.conf_from_file(filename)
 
+@my_bp.route('/')
+def index():
+    return "Hello World"
```
Comment (tzumainn) [inline flocx_market/api/app.py]: Most OpenStack services serve something at the index; for example, if you curl your keystone endpoint, you'll see something like this:

```
[stack@localhost controllers(keystone_admin)]$ curl http://192.168.1.2:5000
{"versions": {"values": [{"status": "stable", "updated": "2019-01-22T00:00:00Z", "media-types": [{"base": "application/json", "type": "application/vnd.openstack.identity-v3+json"}], "id": "v3.12", "links": [{"href": "http://192.168.1.2:5000/v3/", "rel": "self"}]}]}}
```

I'd update this to something similar.

**Comment by tzumainn** [comment]
I'd also suggest changing the commit message from "starting up the app" to "starting up the REST API". Remember, the REST API is just one application that will run from this code; there could be others.

**Comment by tzumainn** [comment]
You probably also want tests for the API. Here's the pecan version: https://github.com/CCI-MOC/esi-leap/blob/master/esi_leap/tests/api/base.py - you may want to look at how keystone tests its API.

**Comment by tzumainn** [comment]
Oh, and the failing CI is because you need a body for your commit message.

**Comment by DanNiESh** [comment]
> I'd also suggest changing the commit message from "starting up the app" to "starting up the REST API". Remember, the REST API is just one application that will run from this code; there could be others.

Sure. But the REST api part is in Filip's branch, not merged in upstream. Should I take some code from Filip?

**Comment by tzumainn** [comment]
I think we need to decide which PR is dependent upon the other. My preference would be for this to get in first, and then for Filip's PR to rebase on top of this.

File: flocx_market/api/app.py [hunk 2]
```diff
@@ -1,20 +1,17 @@
-import pecan
-from pecan import make_app
-from flocx_market import model
-from flocx_market.api import config as api_config
+import os
+from flask import Flask
+from flask import Blueprint
+from flask import jsonify
+my_bp = Blueprint("my_bp", __name__)
```
Comment (tzumainn) [inline flocx_market/api/app.py]: I should have caught this before, but we probably want to rename this to something a bit less casual - flocx_market_api_bp or something

File: flocx_market/api/app.py [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
 
 
-def get_pecan_config():
-    filename = api_config.__file__.replace('.pyc', '.py')   # get the absolute path of the pecan config.py
-    return pecan.configuration.conf_from_file(filename)
+@my_bp.route('/')
+def index():
+    version = {"versions": {"values": [{"status": "in progress", "updated": "2019-07-02T00:00:00Z", "media-types": [{"base": "application/json", "name": "flocx-marketplace"}], "links": [{"href": "http://localhost:8081", "rel": "self"}]}]}}
```
Comment (tzumainn) [inline flocx_market/api/app.py]: This is probably more readable if you put the lines of the dict on different lines. I think we're also calling it 'flocx-market' rather than 'flocx-marketplace'; and instead of hardcoding `http://localhost:8081` we should get it from oslo config.

Comment (DanNiESh) [reply flocx_market/api/app.py]: Should I modify default value for CONF.api.public_endpoint and use it to get the flocx-market url? Then I need to modify some places in config test, since the default value is None.

Comment (tzumainn) [reply flocx_market/api/app.py]: The WSGI service starts up using these parameters, right?

```
            host=CONF.api.host_ip,
            port=CONF.api.port,
```

Just use those.

Comment (DanNiESh) [reply flocx_market/api/app.py]: Thank you! All fixed.

File: flocx_market/api/app.py [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
-    filename = api_config.__file__.replace('.pyc', '.py')   # get the absolute path of the pecan config.py
-    return pecan.configuration.conf_from_file(filename)
+@flocx_market_api_bp.route('/')
+def index():
+    flocx_market_url = CONF.api.host_ip + ':' + str(CONF.api.port)
+    version = {
+        "versions":
+        {"values": [{"status": "in progress",
+         "updated": "2019-07-02T00:00:00Z",
+         "media-types": [{"base": "application/json", "name": "flocx-marketplace"}],
```
Comment (tzumainn) [inline flocx_market/api/app.py]: I think we should rename flocx-marketplace to just flocx-market

**Comment by DanNiESh** [comment]
> Oh, and I think there's a few pep8 complaints that should probably be fixed!

Yes! I'm trying to modify the json pep8 format.

**Comment by DanNiESh** [comment]
> One last minor comment; and I think you need to add Flask to requirements.txt. After that, I think this is good to merge!

requirements.txt, pep8, flocx-marketplace->flocx-market problems fixed.

**Comment by tzumainn** [comment]
Looks good!
